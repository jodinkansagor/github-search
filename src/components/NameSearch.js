import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchUserInfo } from '../actions/searchUserAction';
import { fetchUserRepos } from '../actions/userReposAction';
import'./NameSearch.css';

const NameSearch = () => {

  const [userName, setUserName] = useState('');
  const dispatch = useDispatch();

  const handleSubmit = event => {
    event.preventDefault();
    dispatch(fetchUserInfo(userName));
    dispatch(fetchUserRepos(userName));
  };

  return (
    <form onSubmit={handleSubmit}>
      <h1>Github People Finder!</h1>
      <h3>Enter a Github Username</h3>
      <input type="text" value={userName} onChange={({ target }) => setUserName(target.value)} />
      <button>Submit</button>
    </form>
  )


}
  ;

NameSearch.propTypes = {};

export default NameSearch;