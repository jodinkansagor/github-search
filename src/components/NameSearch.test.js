import React from 'react';
import { shallow } from 'enzyme';
import NameSearch from './NameSearch';
import { Provider } from 'react-redux';
import store from '../store';

describe('Name search component', () => {
  it('render the name search', () => {
    const wrapper = shallow(<Provider store={store}><NameSearch /></Provider>).dive();
    expect(wrapper).toMatchSnapshot();
  });
});