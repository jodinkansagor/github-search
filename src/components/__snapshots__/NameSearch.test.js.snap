// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name search component render the name search 1`] = `<NameSearch />`;
